wget/wget.go

```diff
@@ -0,0 +1,202 @@
+package wget
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+)
+
+// File represents infromation about file
+// we are going to download
+type File struct {
+	io.Reader       // To implement Reader interface
+	totalSize int64 // Total content size in bytes
+	counter   int64 // Total # of bytes transferred
+}
+
+type DownloadManager struct {
+	fileList        map[string]*File // key = link
+	fileListOrder   []string         // to order fileList map iteration
+	fileListLock    sync.Mutex       // to prevent race
+	printFinishedCh chan struct{}    // we use it to wait until last print fucntion will be terminated correctly
+}
+
+// NewDownloadManager is construct function
+func NewDownloadManager() *DownloadManager {
+	return &DownloadManager{
+		fileList:        make(map[string]*File),
+		fileListOrder:   make([]string, 0),
+		printFinishedCh: make(chan struct{}),
+	}
+}
+
+func ParseURL(link string) error {
```

Comment:
What actually dose this function dose? ParseURL but why it returns only error? Need better naming here